### PR TITLE
[PB-1017]:(feature) Use MobileSDK photos processor and gallery

### PR DESCRIPTION
**Pull request by PixoDev**
This PR replaces the Photos JS processor and instead uses the MobileSDK native one. The gallery view has been replaced by a native Android fragment so we avoid layer communication.

### CODE REVIEW

File: src/components/PhotosSyncStatusWidget/index.android.tsx [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+const PhotosSyncStatusWidget: React.FC<PhotosSyncStatusWidgetProps> = (props) => {
+  const navigation = useNavigation<NavigationProp<SettingsStackParamList, 'SettingsHome'>>();
+  const tailwind = useTailwind();
+  const getColor = useGetColor();
+  const { width: windowWidth } = useWindowDimensions();
+  const getProgressWidth = () => {
+    const pendingPhotos = props.pendingSyncs;
+    const syncedPhotos = props.successfulSyncs;
+    if (props.status === PhotosSyncStatus.InProgress || props.status === PhotosSyncStatus.Paused) {
+      const syncedPhotos = props.photosInDevice - props.pendingSyncs;
```
Comment (sg-gs) [inline src/components/PhotosSyncStatusWidget/index.android.tsx:33]: Errored syncs are counted inside pending?

File: src/contexts/Photos/Photos.context.tsx [hunk 2]
```diff
@@ -312,6 +279,16 @@ export const PhotosContextProvider: React.FC = ({ children }) => {
     await asyncStorageService.savePhotosSyncIsEnabled(enable);
 
     setSyncEnabled(enable);
+
+    if (Platform.OS === 'android') {
```
Comment (sg-gs) [inline src/contexts/Photos/Photos.context.tsx:283]: Maybe we can use an enum here, so if it changes, there is no need to replace each check, or a global function

File: src/navigation/PhotosNavigator.tsx [hunk 3]
```diff
@@ -18,7 +20,10 @@ function PhotosNavigator(): JSX.Element {
   useEffect(() => {
     photosCtx.permissions.getPermissionsStatus().then((status) => {
       setPermissionsStatus(status);
-      setIsInitialized(true);
+
+      setTimeout(() => {
```
Comment (sg-gs) [inline src/navigation/PhotosNavigator.tsx:24]: Is this for UX purposes or there is something apart going on here?

File: src/screens/PhotosGalleryScreen/index.android.tsx [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+  const [photosProcessState, setPhotosProcessState] = useState<PhotosProcessProgressUpdate>({
+    status: 'IDLE',
+    failedSyncs: 0,
+    successfulSyncs: 0,
+    totalPhotosInDevice: 0,
+    totalPhotosInRemote: 0,
+    pendingSyncs: 0,
+  });
+
+  /* const checkForPhotosManagerUpdates = async () => {
```
Comment (sg-gs) [inline src/screens/PhotosGalleryScreen/index.android.tsx:41]: Remove this comment please

File: src/screens/PhotosGalleryScreen/index.android.tsx [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+      }
+      const { user, device } = await photos.user.init();
+
+      await internxtMobilePhotosSdk.initialize({
+        photosBucketId: user.bucketId,
+        photosDeviceId: device.id,
+        photosUserId: user.id,
+      });
+
+      await internxtMobilePhotosSdk.startPhotos();
```
Comment (sg-gs) [inline src/screens/PhotosGalleryScreen/index.android.tsx:147]: Are we initializing the SDK only?

File: src/components/PhotosSyncStatusWidget/index.android.tsx (hunk 1 above)
Comment (PixoDev) [reply src/components/PhotosSyncStatusWidget/index.android.tsx:33]: yeah, after all, they are pending syncs

File: src/components/modals/PhotosPreviewInfoModal/index.tsx [hunk 6]
```diff
@@ -11,6 +11,8 @@ import { time } from '@internxt-mobile/services/common/time';
 import AppText from 'src/components/AppText';
 import { PhotosItem } from '@internxt-mobile/types/photos';
 import { PhotosItemActions } from 'src/screens/PhotosPreviewScreen';
+import FastImage from 'react-native-fast-image';
```
[the remark this answers is not in the document]
Comment (PixoDev) [reply src/components/modals/PhotosPreviewInfoModal/index.tsx:14]: indeed, but due to dependencies versions constraints we cannot remove it for now

File: src/contexts/Photos/Photos.context.tsx (hunk 2 above)
Comment (PixoDev) [reply src/contexts/Photos/Photos.context.tsx:283]: this is already a TS union type https://reactnative.dev/docs/platform-specific-code#content

File: src/navigation/PhotosNavigator.tsx (hunk 3 above)
Comment (PixoDev) [reply src/navigation/PhotosNavigator.tsx:24]: is just an UX tweak, so you don't face a white screen when navigating to the Photos gallery

File: src/screens/PhotosGalleryScreen/index.android.tsx (hunk 5 above)
Comment (PixoDev) [reply src/screens/PhotosGalleryScreen/index.android.tsx:147]: WDYM? the Mobile SDK is initialized here: https://github.com/internxt/drive-mobile/blob/8f19c75c484754131f671b17ea23f564bdd58af1/src/store/slices/auth/index.ts#L31

This method only starts the native photos processor

File: index.js [hunk 7]
```diff
@@ -16,6 +16,8 @@ import store from './src/store';
 import 'moment/locale/es';
 import 'moment/locale/en-in';
 import { decode, encode } from 'base-64';
+import { enableScreens } from 'react-native-screens';
```
[the remark this answers is not in the document]
Comment (PixoDev) [reply index.js:19]: I don't find the github issue anymore, but this was causing a problem with the underlying Android Fragment

File: src/screens/PhotosGalleryScreen/index.android.tsx (hunk 5 above)
Comment (sg-gs) [reply src/screens/PhotosGalleryScreen/index.android.tsx:130]: So, the method is called initPhotosMobileSdk but it does init the SDK (L141) and also does init the native photos processor, so the function naming is hiding a part of the behaviour this function has, I think

File: src/contexts/Photos/Photos.context.tsx [hunk 8]
```diff
@@ -159,46 +160,12 @@ export const PhotosContextProvider: React.FC = ({ children }) => {
...
-      }
-    });
-
     initializeSyncIsEnabled().catch((err) => {
       errorService.reportError(err);
     });
 
     return () => {
-      return subscription();
+      // return subscription();
```
[the remark this answers is not in the document]
Comment (sg-gs) [reply src/contexts/Photos/Photos.context.tsx]: The comment is still here

File: src/screens/PhotosGalleryScreen/index.android.tsx (hunk 5 above)
Comment (PixoDev) [reply src/screens/PhotosGalleryScreen/index.android.tsx:130]: I see what you mean now, I will decouple those

File: src/contexts/Photos/Photos.context.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (PixoDev) [reply src/contexts/Photos/Photos.context.tsx]: Maybe you need a refresh:

https://github.com/internxt/drive-mobile/pull/225/files#diff-d8adb9f4a4832cc4925bf991f1005196a2bb2a26aa9c6f6865742ef27a18d5c1L201

[the remark this answers is not in the document]
Comment (sg-gs) [reply src/contexts/Photos/Photos.context.tsx]: Damn cache

**Review by sg-gs — APPROVED** [review]

**Pull request merged**